src/elements/MetricsSelector/MetricsSelector.js

```diff
@@ -119,25 +143,31 @@ const MetricsSelector = ({ maxSelectionNumber, metrics, name, onSelect, preselec
   }, [windowClickHandler, windowScrollHandler, isOpen])
 
   const handleOnChange = selectedMetrics => {
-    onSelect(
-      selectedMetrics.map(metricFullName => {
-        return metrics.find(metric => metric.full_name === metricFullName)
-      })
-    )
+    setPreAppliedMetricsRef(selectedMetrics)
+  }
+
+  const handleApply = () => {
+    const matchedMetrics = preAppliedMetrics.map(metricFullName => {
+      return metrics.find(metric => metric.full_name === metricFullName)
+    })
+    onSelect(matchedMetrics)
+    appliedMetricsRef.current = matchedMetrics
+    setIsOpen(false)
   }
 
   const getSelectValue = (selectedMetrics = []) => {
-    if (isEmpty(selectedMetrics)) {
+    if (isEmpty(appliedMetricsRef.current)) {
```

Comment:
you don't need `selectedMetrics` parameter anymore. Delete it

Reply:
removed